barretenberg/cpp/src/barretenberg/ultra_honk/decider_proving_key.hpp

```diff
@@ -71,6 +75,156 @@ template <class Flavor> class DeciderProvingKey_ {
+                        Polynomial(MAX_DATABUS_SIZE, proving_key.circuit_size);
+                    proving_key.polynomials.return_data_read_tags =
+                        Polynomial(MAX_DATABUS_SIZE, proving_key.circuit_size);
+
+                    // TODO(https://github.com/AztecProtocol/barretenberg/issues/1107): Restricting databus_id to
+                    // databus_size leads to failure.
+                    // const size_t databus_size = std::max({ calldata.size(), secondary_calldata.size(),
+                    // return_data.size() });
+                    proving_key.polynomials.databus_id = Polynomial(proving_key.circuit_size, proving_key.circuit_size);
+                }
+                const size_t max_tables_size =
+                    std::min(static_cast<size_t>(MAX_LOOKUP_TABLES_SIZE), dyadic_circuit_size - 1);
+                size_t table_offset = dyadic_circuit_size - max_tables_size;
+                {
+                    ZoneScopedN("allocating table polynomials");
+                    ASSERT(dyadic_circuit_size > max_tables_size);
+
+                    // Allocate the table polynomials
+                    if constexpr (IsHonkFlavor<Flavor>) {
+                        for (auto& poly : proving_key.polynomials.get_tables()) {
+                            poly = typename Flavor::Polynomial(max_tables_size, dyadic_circuit_size, table_offset);
+                        }
+                    }
+                }
+                {
+                    ZoneScopedN("allocating sigmas and ids");
+                    for (auto& sigma : proving_key.polynomials.get_sigmas()) {
+                        sigma = typename Flavor::Polynomial(proving_key.circuit_size);
+                    }
+                    for (auto& id : proving_key.polynomials.get_ids()) {
+                        id = typename Flavor::Polynomial(proving_key.circuit_size);
+                    }
+                }
+                {
+                    ZoneScopedN("allocating lookup read counts and tags");
+                    // Allocate the read counts and tags polynomials
+                    proving_key.polynomials.lookup_read_counts =
+                        typename Flavor::Polynomial(max_tables_size, dyadic_circuit_size, table_offset);
+                    proving_key.polynomials.lookup_read_tags =
+                        typename Flavor::Polynomial(max_tables_size, dyadic_circuit_size, table_offset);
+                }
+                {
+                    ZoneScopedN("allocating lookup and databus inverses");
+                    // Allocate the lookup_inverses polynomial
+                    const size_t lookup_offset = static_cast<size_t>(circuit.blocks.lookup.trace_offset);
+                    // TODO(https://github.com/AztecProtocol/barretenberg/issues/1033): construct tables and counts
+                    // at top of trace
+                    const size_t table_offset =
```

Comment:
not the nicest set of computations, but necessary I think to handle edge cases of circuit size being too small, etc.... Maybe could move to a function